workers/unity/Packages/io.improbable.gdk.core/Utility/CommandLineUtility.cs

```diff
@@ -80,25 +85,28 @@ public bool TryGetCommandLineValue<T>(string key, ref T value)
             return false;
         }
 
+        private static bool isFlag(string flag)
+        {
+            return flag.StartsWith("+") || flag.StartsWith("-");
+        }
+
         private static Dictionary<string, string> ParseCommandLineArgs(IList<string> args)
         {
             var config = new Dictionary<string, string>();
             for (var i = 0; i < args.Count; i++)
             {
                 var flag = args[i];
-                if (flag.StartsWith("+") || flag.StartsWith("-"))
+                if (isFlag(flag))
                 {
-                    if (i + 1 >= args.Count)
-                    {
-                        throw new ArgumentException(
-                            $"Flag \"{flag}\" requires an argument\nArguments: \"{string.Join(", ", args)}\"");
-                    }
-
-                    var flagArg = args[i + 1];
+                    var isNextArgFlagOrEnd = i + 1 >= args.Count || isFlag(args[i + 1]);
```

Comment:
I think this can be simplified

```csharp
var strippedOfPlus = flag.Substring(1, flag.Length - 1);
if (i + 1 >= args.Count || isFlag(args[i + 1]))
{
    config[strippedOfPlus] = string.Empty;
}
else
{
    config[strippedOfPlus] = args[i + 1];
    i++;
}
```

Something like this